### PR TITLE
Rpi 4 update

### PR DESCRIPTION
**Pull request by Matthew3000**
The support page update for RPI 4 version of the kit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,77 +22,38 @@ Full instruction is provided starting on page 56 of the book. Here are all third
...
-
-Then Thonny reboots and you get the full interface:
-
-![Image](./images/p25-3.png)
-
 ```python
 import RPi.GPIO as GPIO
 import time
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 2]
```diff
@@ -21,77 +22,38 @@ Full instruction is provided starting on page 56 of the book. Here are all third
...
+    print('GPIO cleanup completed.')
 ```
 
 ## 4. Button
 
 ```python
 import RPi.GPIO as GPIO
 import time
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 3]
```diff
@@ -21,77 +22,38 @@ Full instruction is provided starting on page 56 of the book. Here are all third
...
 ```python
 import RPi.GPIO as GPIO
 import time
-
+ 
 GPIO.setmode(GPIO.BCM)
 GPIO.setup(2, GPIO.IN)
-
-while (True):
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 4]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
 GPIO.setup(24, GPIO.OUT)
 ```
 
-## 5. Conditional expressions
+Finally adding exceptions handling:
```
Comment (knopki) [inline README.md]: Missing comma?

File: README.md [hunk 5]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
 ```
 
-## 6. Controlling the leds
+## 6. LED brightness
 
 ```python
 import RPi.GPIO as GPIO
 import time
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 6]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
     else:
         GPIO.output(led, GPIO.LOW)
+ 
 
 button1 = 3
 button2 = 4
 led1 = 14
 led2 = 15
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 7]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
+    print('GPIO cleanup completed.')
 ```
 
 ## 8. Massive optimization
 
 ```python
 import RPi.GPIO as GPIO
 import time
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 8]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
 
 ```python
 import RPi.GPIO as GPIO
 import time
-
+ 
 GPIO.setmode(GPIO.BCM)
 leds = [12, 13, 14, 18]
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 9]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
+ 
 GPIO.setmode(GPIO.BCM)
 leds = [12, 13, 14, 18]
-
+ 
 for led in leds:
     GPIO.setup(led, GPIO.OUT)
     GPIO.output(led, GPIO.HIGH)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 10]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
-
+ 
 for led in leds:
     GPIO.setup(led, GPIO.OUT)
     GPIO.output(led, GPIO.HIGH)
-
+ 
 time.sleep(3)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 11]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
     GPIO.output(led, GPIO.HIGH)
-
+ 
 time.sleep(3)
-
+ 
 for led in leds:
     GPIO.output(led, GPIO.LOW)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 12]
```diff
@@ -111,126 +73,179 @@ Don't forget to add the pin initialization code for the output:
...
-    state = 1-GPIO.input(btn)
+    state = 1 - GPIO.input(btn)
     GPIO.output(led, state)
-
+ 
+ 
 leds = [12, 13, 14, 18]
 buttons = [2, 3, 4, 8]
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 13]
```diff
@@ -243,26 +258,26 @@ wget http://raspberry.amperka.com/web-server.zip -0 web-server.zip
 
 ```python
 from flask import Flask, send_file
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 14]
```diff
@@ -243,26 +258,26 @@ wget http://raspberry.amperka.com/web-server.zip -0 web-server.zip
 
 ```python
 from flask import Flask, send_file
-
+ 
 app = Flask('landingPage')
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 15]
```diff
@@ -243,26 +258,26 @@ wget http://raspberry.amperka.com/web-server.zip -0 web-server.zip
...
-
+ 
 app = Flask('landingPage')
-
+ 
 @app.route('/')
 def index():
     return send_file('landing.html')
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 16]
```diff
@@ -243,26 +258,26 @@ wget http://raspberry.amperka.com/web-server.zip -0 web-server.zip
...
 def index():
     return send_file('landing.html')
-
+ 
 @app.route('/images/<filename>')
 def get_image(filename):
     return send_file('images/'+filename)
-
-app.run(debug=True, port=3000, host='0.0.0.0')
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 17]
```diff
@@ -243,26 +258,26 @@ wget http://raspberry.amperka.com/web-server.zip -0 web-server.zip
...
+app.run(port=3000, host='0.0.0.0')
 ```
 
 Sometimes following steps 8 to 12 you can get console message:  
 `socket.error: [Errno 48] Address already in use`  
 It means that server is still running. In this case use a command:
 
 ```bash
-ps -fA | grep python
+ps -u | grep python
```
Comment (knopki) [inline README.md:280]: Why?

File: README.md [hunk 18]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
 import RPi.GPIO as GPIO
-
-GPIO.setmode(GPIO.BCM)
+ 
 led = 18
+ 
+GPIO.setmode(GPIO.BCM)
 GPIO.setup(led, GPIO.OUT)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 19]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
-```bash
-sudo pip3 install flask-socketio eventlet
-```
-
 ```python
 from flask import Flask, send_file
 from flask_socketio import SocketIO
 import RPi.GPIO as GPIO
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 20]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
 ```
 
-Start the server with command:
-
-```bash
-python3 feedback.py
-```
-
-## 13. Weather widget
+## 13. Weather phidget
```
Comment (knopki) [inline README.md:375]: phi?

File: README.md [hunk 21]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
-    'lon': 'your home city longitude',
-    'units': 'metric',
-    'appid': 'your_key'
+    "lat": "your_city_latitude",
+    "lon": "your_city_longitude",
+    "units": "metric",
+    "appid": "your_key",
 }
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 22]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
+http://raspberry.amperka.com/omx-web.zip
 
 ## Turning on all the leds on the cloud
 
 ```python
 import RPi.GPIO as GPIO
 import time
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 23]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
 
 ```python
 import RPi.GPIO as GPIO
 import time
-
+ 
 GPIO.setmode(GPIO.BCM)
 leds = [10, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 26]
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 24]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
+ 
 GPIO.setmode(GPIO.BCM)
 leds = [10, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 26]
-
+ 
 for led in leds:
     GPIO.setup(led, GPIO.OUT)
     GPIO.output(led, GPIO.HIGH)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 25]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
-
+ 
 for led in leds:
     GPIO.setup(led, GPIO.OUT)
     GPIO.output(led, GPIO.HIGH)
-
+ 
 time.sleep(3)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md [hunk 26]
```diff
@@ -274,145 +289,162 @@ kill -9 xxx
...
     GPIO.output(led, GPIO.HIGH)
-
+ 
 time.sleep(3)
-
+ 
 for led in leds:
     GPIO.output(led, GPIO.LOW)
-
+
```
Comment (knopki) [inline README.md]: `SPC`

File: README.md (hunk 20 above)
Comment (Matthew3000) [reply README.md:375]: phi.
From "physical widget"

Comment (knopki) [reply README.md:375]: Nice.

File: README.md (hunk 17 above)
Comment (Matthew3000) [reply README.md:280]: to lower the quantity of process that show up

File: README.md [hunk 27]
```diff
@@ -321,80 +341,92 @@ import RPi.GPIO as GPIO
...
-```
+- ["BIG BUCK BUNNY"](http://raspberry.amperka.com/Big_Buck_Bunny_1080p.torrent) — a funny blender.io movie
+
+## 17. Movie theater
+
+OMX-web.zip link:
+
+http://raspberry.amperka.com/omx-web.zip
 
 ## Turning on all the leds on the cloud
```
Comment (AlexErAmp) [inline README.md]: You need to delete this item. This is the old code for experiment 8.

**Review by AlexErAmp — APPROVED** [review]
LGTM

**Review by knopki — APPROVED** [review]

**Pull request merged**